### PR TITLE
Run Greengrass samples in CI

### PR DESCRIPTION
**Pull request by sfod**
- Add tests for Greengrass samples: IPC and discovery.
- Add new CI job for Greengrass tests.
- Fix greengrass-discovery sample to iterate over all connections it discovered.
- Add to run_in_ci.py script a possibility to provide data to runnable's stdin.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -641,14 +642,71 @@ jobs:
...
           python3 ${{ env.CI_UTILS_FOLDER }}/run_sample_ci.py --file ${{ env.CI_SAMPLES_CFG_FOLDER }}/ci_run_x509_connect_cfg.json
-      - name: configure AWS credentials (Greengrass)
-        uses: aws-actions/configure-aws-credentials@v2
-        with:
-          role-to-assume: ${{ env.CI_GREENGRASS_ROLE }}
-          aws-region: ${{ env.AWS_DEFAULT_REGION }}
-      - name: run Greengrass Discovery sample
+
+  # Runs the Greengrass samples
+  linux-greengrass-tests:
```
Comment (xiazhvera) [inline .github/workflows/ci.yml:647]: Is there any reason why we didnt run it with the linux matrix?

**Review by xiazhvera — COMMENTED** [review]
Put some questions here. In general the PR looks good to me in. It is impressive to clean up the GG testing workflow here.

File: samples/greengrass/basic_discovery/main.cpp [hunk 2]
```diff
@@ -263,17 +259,53 @@ int main(int argc, char *argv[])
...
+            }
+        };
+
+        connection->Subscribe(cmdData.input_topic.c_str(), AWS_MQTT_QOS_AT_MOST_ONCE, onMessage, onSubAck);
     }
 
-    bool first_input = true;
-    while (true)
+    int cnt = 0;
+    while (++cnt < 3)
```
[the remark this answers is not in the document]
Comment (sfod) [reply samples/greengrass/basic_discovery/main.cpp]: It was specific for CI. But I reverted the sample back to read from stdin, it's simpler this way.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sfod) [reply .github/workflows/ci.yml:647]: This new CI job is based on the `linux-cmoke-tests` job above. By linux matrix you mean specify the image version?

Comment (xiazhvera) [reply .github/workflows/ci.yml:647]: My fault, I thought we ran the samples/tests in different images, which it was actually the `linux-smoke-tests`.

**Review by xiazhvera — APPROVED** [review]

**Pull request merged**